src/main/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepExecution.java

```diff
@@ -329,6 +331,35 @@ private Object readResolve() {
             return getDisplayName();
         }
 
+        @Restricted(NoExternalUse.class) // for Jelly
+        public @CheckForNull String getEnclosingLabel() throws Exception {
+            FlowNode executorStepNode = context.get(FlowNode.class);
+            if (executorStepNode != null) {
+                for (StepExecution exec : executorStepNode.getExecution().getCurrentExecutions(true).get(1, TimeUnit.MINUTES)) {
```

Comment:
We expect `get` to return immediately; the timeout is just to prevent a UI thread from hanging if something weird goes wrong.